From b3be69e65dcc8b84057a8203264a3b07766cd555 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=BCdiger=20K=C3=B6pke?= <Ruediger.Koepke@baslerweb.com>
Date: Tue, 21 Mar 2023 15:17:45 +0100
Subject: [PATCH 1/2] VSI: Fix VIV_VIDEO_EVENT_PASS_JSON usage in
 V4l2Event::eventLoop

The shared memory for the JSON response was wrongly mapped to the offset of the JSON request.
This resulted in an empty JSON response on the client side.
---
 mediacontrol/daemon/V4l2Event.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mediacontrol/daemon/V4l2Event.cpp b/mediacontrol/daemon/V4l2Event.cpp
index bcb4564d5..550cf6b4d 100644
--- a/mediacontrol/daemon/V4l2Event.cpp
+++ b/mediacontrol/daemon/V4l2Event.cpp
@@ -263,7 +263,7 @@ void V4l2Event::eventLoop() {
                 }
                 if (bufMap.find(event.response) == bufMap.end()) {
                     bufMap[event.response] = ::mmap(NULL, VIV_JSON_BUFFER_SIZE,
-                        PROT_READ | PROT_WRITE, MAP_SHARED, eventDevice, event.addr);
+                        PROT_READ | PROT_WRITE, MAP_SHARED, eventDevice, event.response);
                 }
                 Json::Reader reader;
                 std::string str = (char *)bufMap[event.addr];
-- 
2.34.1

